### PR TITLE
[comms, commsdsl] update ports

### PR DESCRIPTION
**Pull request by mathisloge**
**Describe the pull request**

- Update comms and commsdsl to the next version

- Which triplets are supported/not supported? Have you updated the CI baseline?yes

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?yes

### CODE REVIEW

**Comment by mathisloge** [comment]
build failures for static builds maybe related to https://github.com/microsoft/vcpkg/pull/15390 ?

**Comment by JonLiu1993** [comment]
Hi @mathisloge ,Thanks for your PR.
Can you run the command ./vcpkg x-add-version --all --overwrite-version on your local, and then resubmit it? I try to help you modify but I don't have permission to push

**Comment by JonLiu1993** [comment]
@mathisloge ，Please take a look at this：
[install-x64-windows-static-dbg-out.log](https://github.com/microsoft/vcpkg/files/6065755/install-x64-windows-static-dbg-out.log)

**Comment by mathisloge** [comment]
> build failures for static builds maybe related to #15390 ?

@JonLiu1993  i think it might be related to the mentioned PR. 

Since the changes introduced with this PR doesn't change the code related to xml or cmake and the previous version has been build successfully.

**Comment by JonLiu1993** [comment]
@mathisloge ，I agree with you, we can wait for the pr merge and re-run CI

**Comment by mathisloge** [comment]
@JonLiu1993 there was indeed a change in upstream cmake. needed to build only with dynamic linking. 
this is ready to review and merge. the CI failures are not related to this PR (but to this https://github.com/microsoft/vcpkg/pull/17528)

File: ports/comms/vcpkg.json [hunk 1]
```diff
@@ -1,7 +1,6 @@
 {
   "name": "comms",
-  "version-string": "3.1.3",
-  "port-version": 1,
+  "version-string": "3.1.4",
```
Comment (JonLiu1993) [inline ports/comms/vcpkg.json:3]: @mathisloge, we should use "version" instead of "version-string"

File: versions/c-/comms.json [hunk 2]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "df4255fb3f850cf26b583afb6b5c04b47672d4fe",
+      "version-string": "3.1.4",
```
Comment (JonLiu1993) [inline versions/c-/comms.json:5]: we should use "version" instead of "version-string"

File: ports/commsdsl/vcpkg.json [hunk 3]
```diff
@@ -1,11 +1,10 @@
 {
   "name": "commsdsl",
-  "version-string": "3.6.0",
-  "port-version": 1,
+  "version-string": "3.6.1",
```
Comment (JonLiu1993) [inline ports/commsdsl/vcpkg.json:3]: @mathisloge, we should use "version" instead of "version-string"

File: versions/c-/commsdsl.json [hunk 4]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "5a786d6877025b51947f6cdfcfdfecb97f75b913",
+      "version-string": "3.6.1",
```
Comment (JonLiu1993) [inline versions/c-/commsdsl.json:5]: @mathisloge, we should use "version" instead of "version-string"

**Comment by JonLiu1993** [comment]
@mathisloge, please modify the part suggested above, and then we can merge this pr after waiting for the basline pr to be merged

**Comment by mathisloge** [comment]
@JonLiu1993 
have changed it to version-semver`
https://github.com/commschamp/commsdsl#versioning

**Comment by mathisloge** [comment]
@JonLiu1993 merged master after #17528 was merged and now CI is good

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Pull request merged**